src/transport/SecureSessionMgr.cpp

```diff
@@ -119,9 +121,29 @@ Transport::Type SecureSessionMgr::GetTransportType(NodeId peerNodeId)
 CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHeader & payloadHeader,
                                          System::PacketBufferHandle && msgBuf, EncryptedPacketBufferHandle * bufferRetainSlot)
 {
+    PeerConnectionState * state = nullptr;
     PacketHeader unusedPacketHeader;
-    return SendMessage(session, payloadHeader, unusedPacketHeader, std::move(msgBuf), bufferRetainSlot,
-                       EncryptionState::kPayloadIsUnencrypted);
+
+    state = GetPeerConnectionState(session);
+    VerifyOrReturnError(state != nullptr, CHIP_ERROR_NOT_CONNECTED);
+
+    if (!IsControlMessage(payloadHeader) && !state->GetSessionMessageCounter().GetPeerMessageCounter().IsSyncCompleted())
```

Comment:
Per spec, this should only happen for CRMP messages, no?  Why are we doing it in all cases?

Also, why are we doing it for cases when we're not using a group key?  For unicast communication there should be no counter sync going on.

Or am I missing something here and we're somehow restricting this to group comms somewhere?